lib/fs.js

```diff
@@ -591,6 +605,72 @@ function writeSync(fd, buffer, offset, length, position) {
   return result;
 }
 
+// usage:
+// fs.writev(fd, buffer, [chunkLength, [offset, [length, [position]]]],
+//    callback);
+function writev(fd, buffer, chunkLength, offset, length, position, callback) {
+  function wrapper(err, written) {
+    // Retain a reference to chunks so that they can't be GC'ed too soon.
+    callback(err, written || 0, buffer);
+  }
+
+  validateUint32(fd, 'fd');
+
+  const req = new FSReqCallback();
+  req.oncomplete = wrapper;
+
+  callback = maybeCallback(callback || position ||
+     length || offset || chunkLength);
+
+  if (typeof position !== 'number')
+    position = null;
+
+  if (isBuffersArray(buffer)) {
+    return binding.writeBuffers(fd, buffer, position, req);
+  }
+
+  if (typeof chunkLength !== 'number')
+    chunkLength = 1024;
+  if (typeof offset !== 'number')
+    offset = 0;
+  if (typeof position !== 'number')
+    position = null;
+
+  if (isArrayBufferView(buffer) || isTypedArray(buffer)) {
+    const chunks = [];
+    if (typeof length !== 'number')
+      length = buffer.byteLength - offset;
+
+    validateOffsetLengthWrite(offset, length, buffer.byteLength);
+
+    buffer = buffer.slice(offset, offset + length);
+    for (var i = 0; i < buffer.byteLength; i += chunkLength) {
+      chunks.push(buffer.slice(i, i + chunkLength));
+    }
+
+    return binding.writeBuffers(fd, chunks, position, req);
+  }
+
+  if (typeof buffer !== 'string')
+    buffer += '';
+  if (typeof length !== 'number')
+    length = buffer.length - offset;
+
+  buffer = buffer.substring(offset, offset + length);
+
+  // Sliceing the string into chunkLengths and mapping it to array of buffers
+  buffer = buffer
+    .match(new RegExp('[^]{1,' + chunkLength + '}', 'g'))
```

Comment:
That's a wildly inefficient way of slicing a string, especially since you're already using `String#substring()` above (but again, I don't really see the point either.)